### ISSUE TITLE
Question regarding stf deployment using stf commands instead of docker

### ISSUE THREAD

**Issue by xiaweixing** [openstf/stf#412]
## What's the problem (or question)?
I want to deploy STF on multiple Ubuntu machines on which I could not use docker solution.  I have to use stf local, but need detailed instruction on how to do it.
I want to use box1 (10.241.54.10) as the master box, and connect box2(10.241.54.16) to box1, so from http://10.241.54.10:7100/#!/devices, I can see all the devices connected to both box1 and box2.

## What should have happened?
My steps/cmds to run stf:
on box1:
```
start rethinkdb
stf local --allow-remote --public-ip 10.241.54.10 --provider 10.241.54.10 --provider-min-port 15000 --provider-max-port 20000
```
on box2:
```
start rethinkdb
stf local --allow-remote --public-ip 10.241.54.16 --provider 10.241.54.16 --provider-min-port 20000 --provider-max-port 25000
```
Then I can successfully see devices showing up from http://10.241.54.10:7100/#!/devices and http://10.241.54.16:7100/#!/devices
Above steps mean my stf env for both boxes are correct.

I tried to start stf local on box1:
```
stf local --allow-remote --public-ip 10.241.54.10 --provider 10.241.54.10 --provider-min-port 15000 --provider-max-port 20000
```
I checked the processes started by stf local and get the provider one:
```
test     16054 16018  0 Aug30 pts/15   00:01:22 /home/test/.linuxbrew/Cellar/node/6.1.0/bin/node /home/test/.linuxbrew/lib/node_modules/stf/lib/cli.js provider --name 10.241.54.10 --min-port 15000 --max-port 20000 --connect-sub tcp://127.0.0.1:7114 --connect-push tcp://127.0.0.1:7116 --group-timeout 900 --public-ip 10.241.54.10 --storage-url http://localhost:7100/ --adb-host 127.0.0.1 --adb-port 5037 --vnc-initial-size 600x800 --allow-remote
```
then tried to start stf provider on box2(with same port as above):
```
stf provider --allow-remote --name 10.241.54.16 --min-port 20000 --max-port 25000 --connect-sub tcp://10.241.54.10:7114 --connect-push tcp://10.241.54.10:7116 --storage-url http://10.241.54.10
```
but it shows:
INF/provider 11938 [*] Providing all 0 of 5 device(s); ignoring "0123456789ABCDEF", "284cd8fe", "639aa439", "YLHU657DMRG6NNN7", "7TTKM78PNFS8SW7D"
I cannot see devices connected to box2 from http://10.241.54.16:7100/#!/devices

## Help us understand your issue by providing context.

So far, docker solution does not seem to be possible for me to go, so I want to achieve production-like deployment with stf command manually.

## Please tell us details about your environment.

* Are you using `stf local` or a [full deployment](doc/DEPLOYMENT.md): stf local
* Version used (git hash or `stf -V`): 2.0.1
* Environment name and version (e.g. Chrome 39, node.js 5.4): node 6.1.0
* Operating System and version: Ubuntu Server 14.04
* If there was a problem with a specific device, run `adb devices -l` and paste the relevant row here:

**Comment by zangtian2** [comment]
Do you solve the problem?The same as me.

**Comment by xiaweixing** [comment]
No, I finally switched to docker solution.  Suggest you do the same :)

**Comment by zangtian2** [comment]
yeah,i had done it.
However,a new problem.I can see device ,but the device always disapeared and disconnected when i click the device icon.Then it shows "the devices is disconnected".
Form adb devices,the device is inline and it's ok.
So,could you tell me how you deploy stf on multiple Ubuntu machines? Thank you so much.
I had tried many smartphones,make no difference.

**Comment by xiaweixing** [comment]
just follow the deployment guide.  First download docker images on multiple machines, then select one as the main host, create almost all .service file accordingly on it; on other hosts, only need to start provider service.

**Comment by cemakkok** [comment]
Guys, can you share your experience under my post, i need help.

https://github.com/openstf/stf/issues/493

**Comment by sorccu** [comment]
Just stop wasting your time and go with the docker deployment. It's definitely possible to run the commands by yourself without docker, but it would be extremely time consuming to help you with that, so we will not. You can try to take a look at the deployment guide and steal/modify the commands from there.

**Comment by shuoyoung** [comment]
I use  "--connect-sub tcp://127.0.0.1:7114 --connect-push tcp://127.0.0.1:7116" instead of  "--connect-sub tcp://publicip:7114 --connect-push tcp://publicip:7116" in the "stf provider" command and the problem  is solved.

**Comment by habeebshk** [comment]
Hi xiawixing, i am trying to deploy it through docker but having trouble connecting with the provider service. It says that my device is ignored and i don't know what to do,

**Comment by habeebshk** [comment]
i am having a hard time understanding the deployment guide.

**Comment by yuyangcq** [comment]
[](url)> 我使用“ --connect-sub tcp：//127.0.0.1：7114 --connect-push tcp：//127.0.0.1：7116”，而不是“ --connect-sub tcp：// publicip：7114 --connect在“ stf provider”命令中输入-push tcp：// publicip：7116即可解决问题。

我使用命令：stf provider --name localhost.localdomain --min-port 7400 --max-port 7700 --connect-sub tcp://172.50.1.159:7114 --connect-push tcp://172.50.1.159:7116 --group-timeout 900 --public-ip 172.50.1.159 --storage-url http://172.50.1.159:7100/ --adb-host 172.17.244.11 --adb-port 5037 --vnc-initial-size 600x800 --mute-master never --allow-remote

然后就一直停在Receiving input from "tcp://172.50.1.159:7114"，过了几分钟就报错了，这个是什么原因
![2021-05-26_105530](https://user-images.githubusercontent.com/29696080/119603827-884e5d00-be20-11eb-9495-6a61e4b67e5a.png)

**Comment by yuyangcq** [comment]
我使用命令：stf provider --name localhost.localdomain --min-port 7400 --max-port 7700 --connect-sub tcp：//172.50.1.159：7114 --connect-push tcp：//172.50.1.159 ：7116 --group-timeout 900 --public-ip 172.50.1.159 --storage-url http://172.50.1.159:7100/ --adb-host 172.17.244.11 --adb-port 5037 --vnc-initial尺寸600x800-静音大师-不允许远程

然后就一直停在从“ tcp：//172.50.1.159：7114”接收输入，过了几分钟就报错了，这个是什么原因
![2021-05-26_104729](https://user-images.githubusercontent.com/29696080/119603882-aa47df80-be20-11eb-8161-63f9c26a97ad.png)
